core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -588,6 +591,9 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
               if constexpr (ReduceFunctorHasFinal<FunctorType>::value)
                 FunctorFinal<FunctorType, WorkTag>::final(
                     static_cast<const FunctorType&>(functor), results_ptr);
+              if (device_accessible_result_ptr)
+                ValueOps::copy(functor, device_accessible_result_ptr,
+                               &results_ptr[0]);
```

Comment:
Why not directly
```suggestion
              if (m_result_ptr_device_accessible)
                ValueOps::copy(functor, m_result_ptr, &results_ptr[0]);
```

Reply:
`m_result_ptr` might be `nullptr`. Using `device_accessible_result_ptr` instead of `m_result_ptr` everywhere allows me to express that I only care about doing anything with `m_result_ptr` in `workgroup_reduction` (or the corresponding code for `size<=1` here) if it is device accessible.

Reply:
OK